### DISCUSSION TITLE
Check if user has permission to access record at framework level.

### DISCUSSION THREAD

**Discussion by DigNZ** [pocketbase/pocketbase#2701] [Q&A] [answered]
I have a reports endpoint - you send it with a from date, to date and a client to generate the report for. A user has access to multiple clients but not all users can see all clients.

The endpoint checks for valid values for from and to and generates the report. How do I test if the currently authed user has permission for the client they are requesting? I would prefer to leverage the View rule from the api  - if the user can view that client then they can generate and download the report for that client.

Here I'm just testing the client id supplied exists:
```
	_, err := app.Dao().FindRecordById("clients", client)
	if err != nil {
		fmt.Println("Could not find client:", err)
		return c.String(404, "Could not find client")
	}
```

**Comment by ganigeorgiev** [comment, marked as answer]
You can check the example in https://github.com/pocketbase/pocketbase/discussions/2271#discussioncomment-5584359 _(the access rule can be get from the `client.Collection().ViewRule` field)_.

There will be a builtin helper for something like this in the next v0.17/v0.18 release.

**Reply by DigNZ** [reply]
That example even is PDF related! Perfect that will do the job nicely.
